sql/core/src/main/scala/org/apache/spark/sql/jdbc/MsSqlServerDialect.scala

```diff
@@ -30,7 +30,13 @@ private object MsSqlServerDialect extends JdbcDialect {
       // String is recommend by Microsoft SQL Server for datetimeoffset types in non-MS clients
       Option(StringType)
     } else {
-      None
+      val answer = sqlType match {
```

Comment:
`Option(x)` should be `Some(x)` here, though both will work. Same below. You can remove `answer` as redundant.

Reply:
Thanks. Will make a fix. Change to Some(x) will avoid the null check which is redundant in this case.

Reply:
Fixed now.